src/misc/cross.cpp

```diff
@@ -782,6 +783,91 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#ifndef WIN32
+uint16_t xattr_string_to_fattr(const char* xattr)
+{
+	uint16_t fattr = 0;
+
+	if (strchr(xattr, 'R')) {
+		fattr |= DOS_ATTR_READ_ONLY;
+	}
+	if (strchr(xattr, 'H')) {
+		fattr |= DOS_ATTR_HIDDEN;
+	}
+	if (strchr(xattr, 'S')) {
+		fattr |= DOS_ATTR_SYSTEM;
+	}
+	if (strchr(xattr, 'A')) {
+		fattr |= DOS_ATTR_ARCHIVE;
+	}
+
+	return fattr;
+}
+
+void fattr_to_xattr_string(uint16_t fattr, char* outbuf, size_t outsize)
+{
+	memset(outbuf, 0, outsize);
+
+	if (fattr & DOS_ATTR_READ_ONLY) {
+		strcat(outbuf, "R");
+	}
+	if (fattr & DOS_ATTR_HIDDEN) {
+		strcat(outbuf, "H");
+	}
+	if (fattr & DOS_ATTR_SYSTEM) {
+		strcat(outbuf, "S");
+	}
+	if (fattr & DOS_ATTR_ARCHIVE) {
+		strcat(outbuf, "A");
+	}
+}
+
+uint16_t get_fat_attribs_from_xattr(const char *path)
+{
+	char xattr[XATTR_FATTR_LEN];
+#ifdef MACOSX
+	int read_len = getxattr(path, XATTR_FATTR_NAME, xattr, XATTR_FATTR_LEN - 1, 0, 0);
```

Comment:
What are the two zero args at the end? Highly suggest extracting them to meaningfully named constants. We started doing this and it greatly enhances the readability of the code. In this case something like this would be better:

```cpp
constexpr uint32_t position = 0;
constexpr int options       = 0;

const auto num_bytes_read = getxattr(path, FatAttribsExtendedAttributeName,
                                     xattr, FatAttribsExtendedAttributeLength,
                                     position, options);
```